src/lib/core/datetime/date-selection-model.ts

```diff
@@ -22,8 +22,8 @@ export abstract class MatDateSelectionModel<D> implements OnDestroy {
     this.selectionChange.complete();
   }
 
-  /** Adds a date to the current selection. */
-  abstract add(date: D): void;
+  /** Adds a date to the current selection. If silent is set to true, no events are emitted */
+  abstract add(date: D | null, silent?: boolean): void;
```

Comment:
This is actually a bit tricky. Because even if the machine sets it, we kind of need to trigger events to keep everything in sync. There are just occasional places, where I want to avoid endless loops. This means that in some cases I would have to set `userInitiated` to `true` even if it possibly wasn't user initiated.

Reply:
For the purpose of emitting events from `(dateChange)` we need to know which ones were triggered by users, as these events should only be emitted when the change was due to a user interaction. If we want to also have a separate observable that emits all the time, we can do that too and subscribe to the appropriate one in the appropriate place